itou/siaes/management/commands/_import_siae/utils.py

```diff
@@ -182,21 +182,24 @@ def sync_structures(df, source, kinds, build_structure, dry_run):
     print(f"{len(deletable_sirets)} {source} will be deleted when possible.")
 
     # Create structures which do not exist in database yet.
-    siret_to_row = {row.siret: row for _, row in df.iterrows()}
-    for siret in creatable_sirets:
-        row = siret_to_row[siret]
+    for _, row in df[df.siret.isin(creatable_sirets)].iterrows():
+        if not row.auth_email:
+            print(f"{source} siret={row.siret} will not been created as it has no email.")
+            continue
+
```

Comment:
Du coup à l'étape de création des nouvelles structures (EA/GEIQ) tu skippes silencieusement celles sans email, soit. Mais je trouve qu'il manque un check que cela n'arrive pas trop.

Imagine (c'est déjà arrivé plusieurs fois) que demain le fichier EA ait 0 emails sur 900. Le script après ta PR terminera sans bruit et on ne se rendra jamais compte que ça fait des semaines qu'on ne créé plus aucune EA/GEIQ.

Je propose de casser bruyamment si jamais il y a trop de nouvelles structures skippées par manque d'email. Par exemple si plus que 3 ou 5 structures skippées ainsi. Pour bien choisir un chiffre raisonnable, il faut regarder sur l'exemple actuel ce que donnent les 74 emails manquants en pratique une fois qu'on ne regarde plus que les nouvelles structures.

Reply:
Ok donnons sa chance à ton idée alors. Un warning non cassant en fin de script en cas de manque d'emails sur les nouvelles structures. Pourquoi pas.

Reply:
Faut que je reteste avec une DB de prod (et sans limiter les lignes à 10) pour être sûr que tout soit bon mais on aurais ça comme sortie :
- Pour les EA/EATT
```
--------------------------------------------------------------------------------
Rows in file: 950
Rows with a SIRET: 950
Rows after deduplication: 758
Rows with a convention not expired: 736
Rows used: 10
 > With an empty email: 73
 > Creatable: 10
 >> Created: 0
 >> Not created because of missing email: 0
 > Updatable: 1
 >> Updatable: 0
 > Deletable: 1
 >> Deleted: 0
 >> Undeletable: 1
--------------------------------------------------------------------------------
Done.
/app/itou/siaes/management/commands/import_ea_eatt.py:210: UserWarning: Too many missing emails: 73
  warnings.warn(f"Too many missing emails: {info_stats['rows_with_empty_email']}")
/app/itou/siaes/management/commands/import_ea_eatt.py:212: UserWarning: 0 structure(s) could not be created because of missing email
  warnings.warn(
```
- Pour les GEIQ
```
--------------------------------------------------------------------------------
Rows in file: 196
Rows with a SIRET: 192
Rows with an empty email: 1
Rows used: 10
 > Creatable: 10
 >> Created: 0
 >> Not created because of missing email: 0
 > Updatable: 0
 >> Updatable: 0
 > Deletable: 0
 >> Deleted: 0
 >> Undeletable: 0
--------------------------------------------------------------------------------
Done.
/app/itou/siaes/management/commands/import_geiq.py:156: UserWarning: 0 structure(s) could not be created because of missing email
  warnings.warn(f"{info_stats['not_created_because_of_missing_email']} structure(s) could not be created because of missing email")
Method handle took 3.48 seconds to complete
```